### PR TITLE
note on OAuth flow

### PR DESCRIPTION
**Pull request by DaedalusG**
This is a quick addition to the docs to highlight an unusual flow issue that has come up for some customers -- more here:
https://github.com/sourcegraph/sourcegraph/issues/31192

## Test plan
requesting review from other teams

### CODE REVIEW

**Review by indradhanush — APPROVED** [review]
Left comments with questions that I had when reading the note along with some suggestions for typo, grammatical and sentence structuring fixes.

Approve to unblock.

**Comment by indradhanush** [comment]
> ## Test plan
>
> requesting review from other teams

This is a good test plan. :stuck_out_tongue_closed_eyes:

File: doc/admin/auth/index.md [hunk 1]
```diff
@@ -47,6 +47,8 @@ enable sign-in via multiple services. Identities from different providers will b
 Sourcegraph user by comparing the user's verified email address to the email address from the
 external identity provider.
 
+> _NOTE: If OAuth is the only sign-in method avaialble on sign-out a new OAuth sign-in will be immediately attempted upon redirect to the sign-in page. If it is necessary to sign-out and have persistent access to the sign-in page, enable `builtin` sign-in in addition to your OAuth sign-in._
```
[the remark this answers is not in the document]
Comment (DaedalusG) [reply doc/admin/auth/index.md:50]: I don't think this is documented behavior (until this note), I can clarify the meaning of builtin a little further though. One of the reasons I opened the issue on this is that it does seem error prone -- I think that some deployments have been caught in weird cycles due to this style of auth in conjunction with the code around licenses 

@unknwon Any elucidation you can add here?

**Review by unknwon — APPROVED** [review]

File: doc/admin/auth/index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (unknwon) [reply doc/admin/auth/index.md]: Relevant code blocks: https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+If+there+is+only+one+auth+provider+configured&patternType=literal

**Pull request merged**